Update grid3 demo on webkit.org

stwrt opened the pull request:
#### 33f42eefdf559a0376d43d96a68647e28cb14e38
<pre>
Update grid3 demo on webkit.org
<a href="https://bugs.webkit.org/show_bug.cgi?id=304254">https://bugs.webkit.org/show_bug.cgi?id=304254</a>
<a href="https://rdar.apple.com/problem/166621488">rdar://problem/166621488</a>

Reviewed by Brent Fulgham.

Update the css-grid-3 demos to be the new grid-lanes syntax.

* Websites/webkit.org/demos/grid3/megamenu/index.html:
* Websites/webkit.org/demos/grid3/megamenu/styles.css:
(body:has(option[value=&quot;three&quot;]:checked) main nav):
(body:has(option[value=&quot;four&quot;]:checked) main nav):
* Websites/webkit.org/demos/grid3/museum/index.html:
* Websites/webkit.org/demos/grid3/museum/sideways-not.css: Added.
(@layer museum, sideways;):
(h2):
(body:has(option[value=&quot;five&quot;]:checked) main):
(section, footer):
(body:has(option[value=&quot;six&quot;]:checked) main):
(body:has(option[value=&quot;five&quot;]:checked) main,):
(article):
(img):
(section):
(footer):
(article.wide):
(.bird):
(.flowers):
(.brower):
(.sicot):
(.moonlight):
(.niagara):
(.bridge):
(.alen):
* Websites/webkit.org/demos/grid3/museum/sideways-not.html: Copied from Websites/webkit.org/demos/grid3/museum/index.html.
* Websites/webkit.org/demos/grid3/museum/sideways.css: Added.
(@layer museum, sideways;):
(h2):
(body:has(option[value=&quot;five&quot;]:checked) main):
(section, footer):
(body:has(option[value=&quot;six&quot;]:checked) main):
(body:has(option[value=&quot;five&quot;]:checked) main,):
(article):
(img):
(section):
(footer):
(article.wide):
(.bird):
(.flowers):
(.brower):
(.sicot):
(.moonlight):
(.niagara):
(.bridge):
(.alen):
* Websites/webkit.org/demos/grid3/museum/sideways.html: Copied from Websites/webkit.org/demos/grid3/museum/index.html.
* Websites/webkit.org/demos/grid3/museum/styles.css:
(*):
(body):
(:root):
(main):
(body:has(option[value=&quot;one&quot;]:checked) main):
(body:has(option[value=&quot;two&quot;]:checked) main):
(body:has(option[value=&quot;three&quot;]:checked) main):
(article):
(body:has(option[value=&quot;four&quot;]:checked) main):
* Websites/webkit.org/demos/grid3/newspaper/index.html:
* Websites/webkit.org/demos/grid3/newspaper/styles.css:
(body:has(option[value=&quot;three&quot;]:checked) main):
(body:has(option[value=&quot;four&quot;]:checked) main):
* Websites/webkit.org/demos/grid3/photos/images/1552px-Dendrobates_azureus_(Dendrobates_tinctorius)_Edit.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Architecture_ApplePark.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Bee.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Beskid_Wyspowy_-_panorama_z_Jaworzynki.JPG: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Birds.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Boat_passing_under_Elgin_Bridge_during_blue_hour_(2023)-L1003785.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Bow_Bridge_panorama_Central_Park_(86746p).jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Brandslange.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Bridge_CitySkyline.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Bridge_Forest.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Canary_Wharf_from_Limehouse_London_June_2016_HDR.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Caterpillar.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Circle_Shapes.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Crow.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Daslook._Allium_ursinum,_zwellende_bloemknop._18-04-2022_(actm.)_04.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Dawn_in_Puerto_Encantado.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Deer_Meadow.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Dog_GoldenMix.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Faucet.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Flower_Meadow.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Gonepteryx_rhamni_-_Citron-1.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Heilbronn_-_Bockingen_-_Friedhof_-_Blick_von_der_Kapelle_zum_Haupteingang_(1).jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Keylong_West_Lahaul_Himachal_Oct22_A7C_03375_panorama.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Lama_glama_Laguna_Colorada_2.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/MarinaBayAppleStore_Singapore_Texture.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Mekong_bank_with_stilt_dwellings_and_clouds_at_golden_hour_in_Don_Det_Laos.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Mountains_AerialShot.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Mow_Cop_Castle,_Sep_2023.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Nevelzwam_(Clitocybe_nebularis)_24-11-2020._(actm.)_01.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/New York_ CentralPark.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/New York_GeometricShapes.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Ocean_Mountains.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Olympic_Stadium_Munich_-_Rows_of_Seats,_April_2019_-04.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Palacio_de_Schonbrunn,_Viena,_Austria,_2020-02-02,_DD_10.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Parque_Eagle_River,_Anchorage,_Alaska,_Estados_Unidos,_2017-09-01,_DD_02.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Raindrops_on_white_cedar.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/RedwoodNationalParkPlant.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Schloss_Sigmaringen_2022.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/ScrubjaywithAcorn.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Sidewalk_Tunnel.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/SnowcapMountains_Pano.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/StackedBooks.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/StreetAlley_Lights.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Sunset_at_Noronhas&apos;_beach.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Taiwan_Building_1.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Taiwan_Building_3.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/USA-2016-Puerto_Rico-San_Juan-Castillo_San_Felipe_del_Morro_10.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Window_View.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Woods_Foggy.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Wren.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Yellowstone_National_Park_(WY,_USA),_Grand_Prismatic_Spring_--_2022_--_2514.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Yellowstone_National_Park_(WY,_USA),_Lower_Geyser_Basin_--_2022_--_2401.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/index.html:
* Websites/webkit.org/demos/grid3/photos/styles.css:
(body:has(option[value=&quot;three&quot;]:checked) main):
(body:has(option[value=&quot;four&quot;]:checked) main):
(body:has(option[value=&quot;five&quot;]:checked) main):
(body:has(option[value=&quot;ten&quot;]:checked) main):
(body:has(option[value=&quot;six&quot;]:checked) main):
(body:has(option[value=&quot;seven&quot;]:checked) main):
(body:has(option[value=&quot;eight&quot;]:checked)):
(body:has(option[value=&quot;nine&quot;]:checked)):
* Websites/webkit.org/demos/grid3/universal.css:
(*):
(:root):
(img):
(body):
(a):
(a:hover):
(p code):
(main):
(@supports (display: grid-lanes)):
(.warning):
(aside.interface:has(button.hidden)):
(aside.interface):
(nav):
(section.layout-switcher):
(section.number-toggle):
(section.show-code):
(button.hide-controls):
(label):
(input[type=&quot;checkbox&quot;][switch]):
(body:has(option[value=&quot;one&quot;]:checked) .interface pre.one,):
(aside pre.code):
(@supports (grid-template-rows: masonry)):
* Websites/webkit.org/demos/grid3/universal.js:

Canonical link: <a href="https://commits.webkit.org/304559@main">https://commits.webkit.org/304559@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4de4ca48e78846a62e2d5ea91408d97469e7be11

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/135968 "80 style errors") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/159/builds/8329 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/138/builds/47250 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/143675 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/88925 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 ios-apple 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/8995 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/156/builds/8175 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/143675 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/88925 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/60440789-950e-438a-b134-58dd42825001) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/138914 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/6510 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/138/builds/47250 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/143675 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/55bdeac3-e14d-49f4-8f02-4d103cef8437) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/6203 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/138/builds/47250 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe-cairo-libwebrtc~~](https://ews-build.webkit.org/#/builders/166/builds/4277 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/115469 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/138/builds/47250 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/146426 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/153/builds/8013 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/138/builds/47250 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/146426 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/160/builds/8033 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/156/builds/8175 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/146426 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-26-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/138/builds/47250 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/61902 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20938 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/158/builds/8061 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/138/builds/47250 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/157/builds/7783 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/8004 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/152/builds/7864 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
<!--EWS-Status-Bubble-End-->

fantasai commented on Websites/webkit.org/demos/grid3/megamenu/index.html:
```diff
@@ -42,23 +42,19 @@
 }</code></pre>
 </section>
 </aside>
-<footer class="warning"><p>View these demos in <a href="https://developer.apple.com/safari/technology-preview/">Safari Technology Preview</a>, Firefox (after you turn on <a href="https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Experimental_features#masonry_grid_layout">the feature flag</a>), or another browser that 
-<a href="https://caniuse.com/mdn-css_properties_grid-template-rows_masonry">supports Grid Level 3</a>.</p></footer> 
```
We should probably keep this note, just drop the bit about Firefox.

stwrt replied:
Added

fantasai commented on Websites/webkit.org/demos/grid3/museum/index.html:
```diff
@@ -16,8 +16,8 @@
         <option value="one">CSS Grid Level 1</option>
-        <option value="two">Symmetrical columnar layout</option>
-        <option selected value="three">With subgrid and item placement</option>
+        <option selected value="two">Symmetrical columnar layout</option>
+        <option value="three">With subgrid and item placement</option>
```
Was this change intentional? What was the reasoning behind it? @jensimmons

jensimmons replied:
Right now, the With subgrid and item placement demo is the default. Did I change it to Symmetrical columnar layout? I don't remember doing so — but I might have, if the subgrid demo wasn't working correctly. 

If the With subgrid and item placement demo works really well / doesn't seem broken, then let's change it to be the one that loads by default when the page loads… by putting the `selected` attribute on it.

fantasai replied:
Ok, it was intentional. Let's roll with this.

fantasai commented on Websites/webkit.org/demos/grid3/museum/styles.css:
```diff
@@ -229,9 +233,9 @@ body:has(option[value="three"]:checked) main {
    grid-template-columns: 
       repeat(auto-fill, minmax(16ch, 1fr) 8ch);
-   grid-template-rows: masonry;
    grid-auto-flow: dense; /* only applies in browsers with no support for Masonry */
```
This comment is no longer true, so, I wonder what we should do about it.

fantasai commented on Websites/webkit.org/demos/grid3/photos/index.html:
```diff
@@ -48,52 +48,45 @@
-  display: grid;
-  grid-template-columns: masonry;
+  display: grid-lanes;
+  grid-auto-flow: column;
```
You need to remove this line.

```suggestion
```

fantasai commented on Websites/webkit.org/demos/grid3/photos/styles.css:
```diff
@@ -217,10 +210,10 @@ body:has(option[value="eight"]:checked) {
 		gap: 1rem;
 		grid-template-rows: repeat(5, 150px);
-		grid-template-columns: masonry;
+		grid-auto-flow: column;
```
Also remove this line.
```suggestion
```

brentfulgham reviewed (approved):
rs=me

stwrt committed:
Update grid3 demo on webkit.org
https://bugs.webkit.org/show_bug.cgi?id=304254
rdar://problem/166621488

Reviewed by Brent Fulgham.

Update the css-grid-3 demos to be the new grid-lanes syntax.

* Websites/webkit.org/demos/grid3/megamenu/index.html:
* Websites/webkit.org/demos/grid3/megamenu/styles.css:
(body:has(option[value="three"]:checked) main nav):
(body:has(option[value="four"]:checked) main nav):
* Websites/webkit.org/demos/grid3/museum/index.html:
* Websites/webkit.org/demos/grid3/museum/sideways-not.css: Added.
(@layer museum, sideways;):
(h2):
(body:has(option[value="five"]:checked) main):
(section, footer):
(body:has(option[value="six"]:checked) main):
(body:has(option[value="five"]:checked) main,):
(article):
(img):
(section):
(footer):
(article.wide):
(.bird):
(.flowers):
(.brower):
(.sicot):
(.moonlight):
(.niagara):
(.bridge):
(.alen):
* Websites/webkit.org/demos/grid3/museum/sideways-not.html: Copied from Websites/webkit.org/demos/grid3/museum/index.html.
* Websites/webkit.org/demos/grid3/museum/sideways.css: Added.
(@layer museum, sideways;):
(h2):
(body:has(option[value="five"]:checked) main):
(section, footer):
(body:has(option[value="six"]:checked) main):
(body:has(option[value="five"]:checked) main,):
(article):
(img):
(section):
(footer):
(article.wide):
(.bird):
(.flowers):
(.brower):
(.sicot):
(.moonlight):
(.niagara):
(.bridge):
(.alen):
* Websites/webkit.org/demos/grid3/museum/sideways.html: Copied from Websites/webkit.org/demos/grid3/museum/index.html.
* Websites/webkit.org/demos/grid3/museum/styles.css:
(*):
(body):
(:root):
(main):
(body:has(option[value="one"]:checked) main):
(body:has(option[value="two"]:checked) main):
(body:has(option[value="three"]:checked) main):
(article):
(body:has(option[value="four"]:checked) main):
* Websites/webkit.org/demos/grid3/newspaper/index.html:
* Websites/webkit.org/demos/grid3/newspaper/styles.css:
(body:has(option[value="three"]:checked) main):
(body:has(option[value="four"]:checked) main):
* Websites/webkit.org/demos/grid3/photos/images/1552px-Dendrobates_azureus_(Dendrobates_tinctorius)_Edit.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Architecture_ApplePark.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Bee.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Beskid_Wyspowy_-_panorama_z_Jaworzynki.JPG: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Birds.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Boat_passing_under_Elgin_Bridge_during_blue_hour_(2023)-L1003785.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Bow_Bridge_panorama_Central_Park_(86746p).jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Brandslange.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Bridge_CitySkyline.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Bridge_Forest.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Canary_Wharf_from_Limehouse_London_June_2016_HDR.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Caterpillar.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Circle_Shapes.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Crow.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Daslook._Allium_ursinum,_zwellende_bloemknop._18-04-2022_(actm.)_04.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Dawn_in_Puerto_Encantado.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Deer_Meadow.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Dog_GoldenMix.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Faucet.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Flower_Meadow.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Gonepteryx_rhamni_-_Citron-1.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Heilbronn_-_Bockingen_-_Friedhof_-_Blick_von_der_Kapelle_zum_Haupteingang_(1).jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Keylong_West_Lahaul_Himachal_Oct22_A7C_03375_panorama.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Lama_glama_Laguna_Colorada_2.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/MarinaBayAppleStore_Singapore_Texture.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Mekong_bank_with_stilt_dwellings_and_clouds_at_golden_hour_in_Don_Det_Laos.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Mountains_AerialShot.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Mow_Cop_Castle,_Sep_2023.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Nevelzwam_(Clitocybe_nebularis)_24-11-2020._(actm.)_01.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/New York_ CentralPark.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/New York_GeometricShapes.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Ocean_Mountains.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Olympic_Stadium_Munich_-_Rows_of_Seats,_April_2019_-04.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Palacio_de_Schonbrunn,_Viena,_Austria,_2020-02-02,_DD_10.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Parque_Eagle_River,_Anchorage,_Alaska,_Estados_Unidos,_2017-09-01,_DD_02.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Raindrops_on_white_cedar.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/RedwoodNationalParkPlant.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Schloss_Sigmaringen_2022.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/ScrubjaywithAcorn.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Sidewalk_Tunnel.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/SnowcapMountains_Pano.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/StackedBooks.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/StreetAlley_Lights.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Sunset_at_Noronhas'_beach.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Taiwan_Building_1.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Taiwan_Building_3.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/USA-2016-Puerto_Rico-San_Juan-Castillo_San_Felipe_del_Morro_10.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Window_View.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Woods_Foggy.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Wren.jpeg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Yellowstone_National_Park_(WY,_USA),_Grand_Prismatic_Spring_--_2022_--_2514.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/images/Yellowstone_National_Park_(WY,_USA),_Lower_Geyser_Basin_--_2022_--_2401.jpg: Removed.
* Websites/webkit.org/demos/grid3/photos/index.html:
* Websites/webkit.org/demos/grid3/photos/styles.css:
(body:has(option[value="three"]:checked) main):
(body:has(option[value="four"]:checked) main):
(body:has(option[value="five"]:checked) main):
(body:has(option[value="ten"]:checked) main):
(body:has(option[value="six"]:checked) main):
(body:has(option[value="seven"]:checked) main):
(body:has(option[value="eight"]:checked)):
(body:has(option[value="nine"]:checked)):
* Websites/webkit.org/demos/grid3/universal.css:
(*):
(:root):
(img):
(body):
(a):
(a:hover):
(p code):
(main):
(@supports (display: grid-lanes)):
(.warning):
(aside.interface:has(button.hidden)):
(aside.interface):
(nav):
(section.layout-switcher):
(section.number-toggle):
(section.show-code):
(button.hide-controls):
(label):
(input[type="checkbox"][switch]):
(body:has(option[value="one"]:checked) .interface pre.one,):
(aside pre.code):
(@supports (grid-template-rows: masonry)):
* Websites/webkit.org/demos/grid3/universal.js:

Canonical link: https://commits.webkit.org/304559@main

yisibl commented on Websites/webkit.org/demos/grid3/megamenu/styles.css:
```diff
@@ -73,18 +73,16 @@ body:has(option[value="two"]:checked) main nav {
-	display: grid;
+	display: grid-lanes;
 	grid-template-columns: repeat(auto-fill, minmax(24ch, 1fr));
-	grid-template-rows: masonry;
```
Can you preserve the old CSS using `@supports`? Otherwise, older versions of Safari won't display the masonry layout. When some users keep macOS on older versions, Safari won't upgrade to the latest version, resulting in compatibility loss.

jensimmons replied:
@yisibl The syntax for the earlier specification for Grid Lanes only shipped in Safari Technology Preview and in Firefox behind a flag — never in Safari, and never for regular users on Firefox. 

The purpose of these demos is to teach best practices. It's not a best practice to add `@supports` code for an early draft of Grid 3 that didn't ship to users. 

While providing fallbacks that use `grid-template-rows: masonry` would keep these demos running for the tiny number of people who are running an old version of Safari Technology Preview because they have not yet updated their Mac to macOS Tahoe _or macOS Sequoia_ (either works), that benefit doesn't seem worth making the code harder to read and understand for the vast majority of developers. 

It's best to move on from the early ideas of what this layout system could have been, and only teach the final version of the spec. There are already far too many demos being made that teach syntax that was never resolved on, that was never part of the official specification. Such chaos only makes it confusing for developers. In the past, when a feature was taught over & over using out of date or never official syntax, it seriously hurt adoption of the feature long term. It took almost a decade for Flexbox to recover, for example.

yisibl replied:
Alright, thank you for your explanation. That makes sense.
